### ISSUE TITLE
SBT CalibanClient plugin & Client code autogen

### ISSUE THREAD

**Issue by dylanraithel** [ghostdogpr/caliban#1058]
I've got a graphql service stood up and would like to ingest the schema to generate a client I can build on.

I'm using caliban-client 1.1.1

I've followed the docs here: https://ghostdogpr.github.io/caliban/docs/client.html#from-a-server-url

and attempted to have sbt hit a /graphql endpoint to fetch the schema but am getting the error below.

```
sttp.client3.SttpClientException$ConnectException: Exception when sending request: POST https://foo/graphql
sttp.client3.SttpClientException$.defaultExceptionToSttpClientException(SttpClientException.scala:33)
sttp.client3.asynchttpclient.AsyncHttpClientBackend.$anonfun$adjustExceptions$1(AsyncHttpClientBackend.scala:227)
sttp.client3.SttpClientException$$anonfun$adjustExceptions$1.applyOrElse(SttpClientException.scala:57)
sttp.client3.SttpClientException$$anonfun$adjustExceptions$1.applyOrElse(SttpClientException.scala:56)
zio.ZIO.$anonfun$catchSome$1(ZIO.scala:414)
scala.util.Either.fold(Either.scala:192)
zio.ZIO.tryRescue$1(ZIO.scala:414)
zio.ZIO.$anonfun$catchSome$6(ZIO.scala:416)
zio.internal.tracing.ZIOFn$$anon$1.apply(ZIOFn.scala:46)
zio.internal.FiberContext.evaluateNow(FiberContext.scala:914)
zio.internal.FiberContext.$anonfun$evaluateLater$1(FiberContext.scala:776)
java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
```

https://github.com/ghostdogpr/caliban/blob/12140947ae10144249e35a6de956448eb64e0910/codegen-sbt/src/main/scala/caliban/codegen/CalibanSourceGenerator.scala#L85-L97

What might be causing this error? This part of the CalibanSourceGenerator looks like it underwent substantial changes recently but that change merged into master hasn't gone into a tagged release yet. Please kindly advise, thanks!

BTW: I've tweaked sbt and logback settings a few different ways to get more info from error but haven't been able to get more of a stack trace than what's above. I checked the docs and didn't see any references to logback / logging settings for Caliban or the caliban-client.

**Comment by ghostdogpr** [comment]
When you generate code from URL, the caliban sbt plugin will call the provided URL to get the schema by introspection (see https://github.com/ghostdogpr/caliban/blob/1add28725345c83e7a89a561a398c93166fc2e82/tools/src/main/scala/caliban/tools/IntrospectionClient.scala#L26). Looking at the error, it looks like the URL is not responding.

An easy way to test this is to add a dependency to `caliban-tools`, which allows you calling the code above directly, or calling the code generation (https://github.com/ghostdogpr/caliban/blob/1add28725345c83e7a89a561a398c93166fc2e82/tools/src/main/scala/caliban/tools/Codegen.scala#L18).

**Comment by dylanraithel** [comment]
Thanks @ghostdogpr it does appear to be an issue on our side with the /graphql endpoint. When I POST using curl I get similar behavior.

I did try your recommendation and wound up down a rabbit hole of ZIO chains and wrappers. I've generally found that a lot of the functional libraries have a high barrier to entry and deviating at all from library code to do simple things like printing exception messages can become quite challenging.

That said, I like that this project has momentum and you are providing examples, so thank you! Some other references I perused and tried to run manually in an SBT Scala console are from your example app. ZIO didn't want to play nice and a lot of their docs are 404, so I've decided to just use the project scoped file loader for now (src/main/graphql/foo.grapql)

Let's close this issue, and once I figure out what's wrong with the /graphql endpoint I'll open a new one if needed.

https://github.com/ghostdogpr/caliban/blob/2c8c71c29a4a87dc064f90b111cf74fa359af668/examples/src/main/scala/example/stitching/ExampleApp.scala#L12


Thanks!
